api/v1alpha1/hosted_controlplane.go

```diff
@@ -25,15 +25,17 @@ type HostedControlPlane struct {
 
 // HostedControlPlaneSpec defines the desired state of HostedControlPlane
 type HostedControlPlaneSpec struct {
-	ReleaseImage  string                      `json:"releaseImage"`
-	PullSecret    corev1.LocalObjectReference `json:"pullSecret"`
-	ServiceCIDR   string                      `json:"serviceCIDR"`
-	PodCIDR       string                      `json:"podCIDR"`
-	MachineCIDR   string                      `json:"machineCIDR"`
-	SSHKey        corev1.LocalObjectReference `json:"sshKey"`
-	ProviderCreds corev1.LocalObjectReference `json:"providerCreds"`
-	InfraID       string                      `json:"infraID"`
-	Platform      PlatformSpec                `json:"platform"`
+	ReleaseImage                             string                      `json:"releaseImage"`
+	PullSecret                               corev1.LocalObjectReference `json:"pullSecret"`
+	ServiceCIDR                              string                      `json:"serviceCIDR"`
+	PodCIDR                                  string                      `json:"podCIDR"`
+	MachineCIDR                              string                      `json:"machineCIDR"`
+	SSHKey                                   corev1.LocalObjectReference `json:"sshKey"`
+	ProviderCreds                            corev1.LocalObjectReference `json:"providerCreds"`
+	InfraID                                  string                      `json:"infraID"`
+	Platform                                 PlatformSpec                `json:"platform"`
+	ControlPlaneServiceTypeStrategy          string                      `json:"controlPlaneServiceTypeStrategy,omitempty"`
```

Comment:
As with everything in programming, naming is hard.

"ControlPlane" is implied by being in the `HostedControlPlaneSpec` in my mind.

I think we could just go with `ServiceType` and `ServiceAddress`, `Address` leaving room for both a DNS name or an IP address.

We should avoid "Ingress" in the name due to it implying application layer routing when these are protocol layer.

Reply:
Renamed!